unity-renderer/Assets/Scripts/MainScripts/DCL/WorldRuntime/ECSRuntime/ComponentsManager/ECSComponentsManager.cs

```diff
@@ -53,19 +54,21 @@ public IECSComponent GetOrCreateComponent(int componentId)
         /// <returns>the instance of existing or newly created component</returns>
         public IECSComponent GetOrCreateComponent(int componentId, IParcelScene scene, IDCLEntity entity)
         {
-            if (loadedComponents.TryGetValue(componentId, out IECSComponent component))
+            IECSComponent component = GetOrCreateComponent(componentId);
+            if (component == null || component.HasComponent(scene, entity))
```

Comment:
I find a bit backwards that a `component` has a method `.HasComponent`.

I would rename it to `BelongsTo`  `IsContainedIn`  `IsPartOfScene`... or move it to a Utils static method.

Reply:
yeah, I'm not liking it either, I'm planning on changing all those components method's name, I'm not sure about the name "component" either since is more a collection than a component